test/ruby/test_gc_compact.rb

```diff
@@ -354,7 +352,7 @@ def test_moving_arrays_up_size_pools
 
   def test_moving_objects_between_size_pools
     omit if GC::INTERNAL_CONSTANTS[:SIZE_POOL_COUNT] == 1
-    pend "Flaky on some platforms" if /solaris|mswin/i =~ RUBY_PLATFORM
+    omit "Flaky on Solaris" if /solaris/i =~ RUBY_PLATFORM
```

Comment:
Still flaky on Solaris?

Reply:
The issue on Solaris is different. The issue on Solaris is:

```
TestGCCompact#test_moving_objects_between_size_pools [/export/home/users/chkbuild/cb-gcc/tmp/build/20230731T230003Z/ruby/test/ruby/test_gc_compact.rb:378]:
Expected 499 to be >= 500.
```

This means that the object was not moved at all, which this PR does not fix. This PR fixes when the object is moved, but is moved back due to reverting the stack. I haven't debugged why this test is flaky on Solaris.

My suspicion is that the last object being created remains in the stack or in a register on Solaris causing the object to be pinned, which I don't know how to fix.